---
title: "Untitled"
author: "Annika Cleven"
date: "2024-11-20"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Project Motivation

Our project is motivated by "Epigenetic dysregulation in Alzheimerâ€™s disease peripheral immunity" (2024, Ramakrishnan).  This study investigates the transcriptional changes in the peripheral immune system in Alzheimer's disease (AD), using advanced single-cell sequencing techniques such as scATAC-seq and scRNA-seq. Ramakrishnan and his team found a considerable amount of open chromatin in the peripheral immune cells in AD. Also, to the researcher's surprise, they also identified differentiallly accessible chromatin regions in gene associated with the sporadic AD risk.  They determined this by checking for overlap between the differentially expressed genes and the differentially accessible regions. This is significant as these genes may be associated risk for sporadic AD.       

In the investigation, the researchers include a list of the differentially expressed genes (DEGs), the DEG sites, their associated log 2 fold change, and overlaps between these DEGs and differentially accessible regions (DARs). While they provide overlaps between the DARs and DEGs, they do not give any further information. In our analysis we dive into detail about the distance related with the DAR and DEG overlap.  After identifying It could be potentially biologically interesting to know the distance and direction between the DEG and the nearest gene in the DAR. A primary goal of our investigation is to verify the study's binary overlap results and quantify the distance to and direction from the nearest gene where the overlap occurs. The researchers have also annotated, albeit in a very elementary manner, the DEGs' cell type, as well as the type of the nearest gene. We can also look for trends between gene and cell type and distance to the nearest gene. 

## Methods

The study began by provided differentially accessible regions and differentially expressed genes. Using the differentially expressed gene names they provided and annotated the gene coordinates that correspond with the gene using the `EnsDb.Hsapiens.v86` library. Using the two GRanges objects containing the DARs and DEGs we were interested in quantifying the overlap differentially expressed genes and the differentially accessible regions. Here we are particularly interested in the promoter regions of the DEGs and we can extract those from our GRanges object using the promoters() function. 

```{r, eval=FALSE}
# Define promoter regions for DEGs (e.g., 2 kb upstream and 500 bp downstream of the TSS)
degs_promoters <- promoters(degs_gr, upstream = 2000, downstream = 500)
```



